nixos/modules/installer/tools/nixos-rebuild.sh

```diff
@@ -440,17 +441,18 @@ if [ "$action" = list-generations ]; then
         kernel_version="$(ls "$kernel_dir/lib/modules")"
 
         build_date="$(date --date="@$(stat "$generation_dir" --format=%W)" "+%a %F %T")"
+        if [ "$generation_number" = "$current_generation" ]; then
+            current="  (current)"
+        fi
 
-        echo "$generation_number,$nixos_version,$kernel_version,$build_date"
+        echo "$generation_number,$nixos_version,$kernel_version,$build_date$current"
     }
 
-    description=""
+    declare -a description
     for generation_dir in /nix/var/nix/profiles/system-*-link ; do
-        description="$description\n$(describe_generation "$generation_dir")"
+        description+=("$(describe_generation "$generation_dir")")
     done
-    current_generation="$(generation_from_dir $(readlink /nix/var/nix/profiles/system))"
-    echo -e "$description" |
-        sed 's/^\<'"$current_generation"'\>.*/&  (current)/' | # add current generation tag to line
```

Comment:
I also removed the `sed` in favour of the `describe_generation` echoing the "(current)", because I think it is much more obvious what hapens